src/faction_camp.cpp

```diff
@@ -306,7 +306,7 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
     std::string bldg = bcp->next_upgrade( base_dir );
 
     if( bldg != "null" ) {
-        comp_list npc_list = companion_list( p, "_faction_upgrade_camp" );
+        comp_list npc_list = talk_function::companion_list( p, "_faction_upgrade_camp" );
```

Comment:
add 
```
comp_list basecamp::companion_list( const std::string mission_id )
{
    return talk_function::companion_list( pos, "FACTION_CAMP", mission_id );
}
```suggestion
        comp_list npc_list = talk_function::companion_list( pos, "FACTION_CAMP", "_faction_upgrade_camp" );
```